src/hotspot/share/opto/loopopts.cpp

```diff
@@ -1685,7 +1689,8 @@ void PhaseIdealLoop::try_sink_out_of_loop(Node* n) {
       !n->is_OpaqueNotNull() &&
       !n->is_OpaqueInitializedAssertionPredicate() &&
       !n->is_OpaqueTemplateAssertionPredicate() &&
-      !n->is_Type()) {
```

Comment:
I cannot remember exactly, how often was it a problem without JDK-8349479? If it was more common, we might want to only allow it when `KillPathsReachableByDeadTypeNode` is set.

Reply:
I made that change.
As far as I remember, the logic removed by JDK-8319372 played a key role in those failures. Not sure if any were still reproducible after than one.